sensor/kubernetes/listener/resources/convert.go

```diff
@@ -362,11 +373,41 @@ func (w *deploymentWrap) populateImageMetadata(localImages set.StringSet, pods .
 				image.Id = digest
 				image.NotPullable = !imageUtils.IsPullable(c.ImageID)
 				image.IsClusterLocal = localImages.Contains(image.GetName().GetFullName())
+				updateImageWithNewerImageName(image, runtimeImageName, false)
 			}
 		}
 	}
 }
 
+// updateImageWithNewerImageName will update the registry, remote, and full name
+// of image from a newer image name, such as the one provided by the container
+// runtime. This is needed in order to support CRI-O's unqualified search registries
+// and short name aliases, for more details refer to:
+// https://github.com/containers/image/blob/main/docs/containers-registries.conf.5.md
+//
+// The image in a workload's spec may differ from what the container runtime pulls,
+// therefore the image name must be updated to accurately reflect the pulled image
+// to give downstream processes (like scanning) a better chance at completing.
+func updateImageWithNewerImageName(image *storage.ContainerImage, newerImageName *storage.ImageName, updDigest bool) {
+	if !features.UnqualifiedSearchRegistries.Enabled() || image.GetName() == nil || newerImageName == nil {
+		return
+	}
+
+	imageName := image.GetName()
+	origFullName := imageName.GetFullName()
+	if imageName.GetRegistry() != newerImageName.GetRegistry() || imageName.GetRemote() != newerImageName.GetRemote() {
```

Comment:
just a sanity check: only the registry/"remote" can be changed, right? The tag/digest are not touched?

Reply:
That's right, and that is also the reason why both `NormalizeImageFullName` and `NormalizeImageFullNameNoSha` are used - to ensure the tag/digest remain untouched given what was provided.